src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -79,6 +80,11 @@ public class JsonMessageFormatter : IJsonRpcMessageTextFormatter
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
         private Encoding encoding;
 
+        private long nextProgressId = 0;
+        private object requestId = null;
+        private Dictionary<object, long> requestProgressMap = new Dictionary<object, long>();
```

Comment:
>requestProgressMap [](start = 41, length = 18)

Could there be multiple threads accessing these Dictionaries at the same time?

Reply:
This is not the latest update, this issue has been addressed already by adding locks when using the dictionaries.

---
In reply to: [304568057](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r304568057) [](ancestors = 304568057)